File: lab5/main.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
  int n;
  cin >> n;
  for (int i=2; (pow(2, i)-1) <= n; i++) {
      int amount_of_divisors = 0;
      int current_number = pow(2, i) - 1;
      int j = 1;
      for (j; j<=current_number;j++) {
          if(current_number%j==0) {
              amount_of_divisors++;
          }
      }
      if (amount_of_divisors==2){
          cout << current_number << endl;
      }
  }
    
  return 0;
}

File: lab5/main.py
n = int(input())
i = 2
while(2**i-1 <=n):
    current_number = 2**i-1
    amount_of_divisors = 0
    for j in range(1, current_number+1):
        if current_number%j==0:
            amount_of_divisors+=1
    if amount_of_divisors == 2:
        print(current_number) 
    i+=1 